### PR TITLE
#15 - CRUD get all systems

### PR DESCRIPTION
**Pull request by leoleader**
## Changes

added service function for get all systems

## Notes

currently just returning result from findMany as json, not entirely sure if that's the desired
way to return resulting data for this so would like feedback

## Test Cases

- Haven't decided on unit testing for this yet

## To Do

- [] Unit testing (maybe)

### CODE REVIEW

File: scylla-server/src/index.ts [hunk 1]
```diff
@@ -20,6 +20,18 @@ app.get("/runs", async (req: Request, res: Response) => {
   }
 });
 
+// #15 CRUD Operations 
+// Service Function, gets all systems and returns result as json
+app.get("/systems", async (req: Request, res: Response) => {
```
Comment (Peyton-McKee) [inline scylla-server/src/index.ts:25]: ok so this would be great if we were still using http. But now that were using a socket we actually just need the prisma.system.findMany() call in an exported function in  a system.services.ts file in a services folder.

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

File: scylla-server/src/index.ts (hunk 1 above)
Comment (leoleader) [reply scylla-server/src/index.ts]: ok i've updated so it should work for websocket

File: scylla-server/src/services/systems.services.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// CRUD operation to get all systems with ResponseFunction type
+
+export const getAllSystems: ResponseFunction = async (data: JSON) => {
+  try {
+    const data = await prisma.system.findMany();
+    // was getting type errors trying to return just data
+    // so added JSON parse and stringify to make it conform to JSON
+    return JSON.parse(JSON.stringify(data));
+  } catch (error) {
+    throw new Error('Error getting all systems');
```
Comment (Peyton-McKee) [inline scylla-server/src/services/systems.services.ts:13]: you shouldnt need a try catch here

File: scylla-server/src/utils/message-maps.utils.ts [hunk 3]
```diff
@@ -7,7 +11,7 @@ export const createServerMessageMap = (): Map<string, ResponseFunction> => {
   const serverMessageMap = new Map<string, ResponseFunction>();
   serverMessageMap.set('test', (data: JSON) => {
     console.log('test', data);
-    return data;
+    return Promise.resolve(data);
```
Comment (Peyton-McKee) [inline scylla-server/src/utils/message-maps.utils.ts:14]: why did you add this?

File: scylla-server/src/utils/message-maps.utils.ts [hunk 4]
```diff
@@ -20,7 +24,10 @@ export const createClientMessageMap = (): Map<string, ResponseFunction> => {
   const clientMessageMap = new Map<string, ResponseFunction>();
   clientMessageMap.set('test', (data: JSON) => {
```
Comment (Peyton-McKee) [inline scylla-server/src/utils/message-maps.utils.ts:25]: you can delete this now, it was just a template for what it would look like

File: scylla-server/src/services/systems.services.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import prisma from '../prisma/prisma-client';
+import { ResponseFunction } from '../utils/message-maps.utils';
+
+// CRUD operation to get all systems with ResponseFunction type
+
```
Comment (Peyton-McKee) [inline scylla-server/src/services/systems.services.ts:5]: use JavaDoc i.e /** */ it should prompt you with paramter values when you do the autocomplete with VSCode

File: scylla-server/src/utils/message-maps.utils.ts [hunk 6]
```diff
@@ -20,7 +24,10 @@ export const createClientMessageMap = (): Map<string, ResponseFunction> => {
...
     console.log('test', data);
-    return data;
+    return Promise.resolve(data);
   });
+  // adds the getAllSystems CRUD to ClientMessageMap
+  clientMessageMap.set('getAllSystems', getAllSystems);
+  console.log('getAllSystems', getAllSystems);
```
Comment (Peyton-McKee) [inline scylla-server/src/utils/message-maps.utils.ts:31]: you can delete this console log

File: scylla-server/src/services/systems.services.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+export const getAllSystems: ResponseFunction = async (data: JSON) => {
```
Comment (Peyton-McKee) [inline scylla-server/src/services/systems.services.ts:6]: since were not using data in this particular function (its undefined) you can preface it like this _data so that the linting thing goes away

File: scylla-server/src/services/systems.services.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  try {
+    const data = await prisma.system.findMany();
+    // was getting type errors trying to return just data
+    // so added JSON parse and stringify to make it conform to JSON
+    return JSON.parse(JSON.stringify(data));
```
Comment (Peyton-McKee) [inline scylla-server/src/services/systems.services.ts:11]: ok i didnt realize you would have to do this, so instead can you change the ResponseFunction type to return a string rather than a JSON Object and then just return JSON.stringify(data)

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Lit Job! Left a few comments.

File: scylla-server/tests/services.test.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import prisma from '../src/prisma/prisma-client';
```
Comment (Peyton-McKee) [inline scylla-server/tests/services.test.ts:1]: can you specify this file even more to be systems-services.test.ts

**Review by Peyton-McKee — APPROVED** [review]
WWWWW
